### PR TITLE
API Documentation

### PR DESCRIPTION
**Pull request by JonaszJestem**

### CODE REVIEW

File: src/core/index.ts [hunk 1]
```diff
@@ -3,11 +3,18 @@ import config from '../config';
 import { imageController } from './image/image.controller';
 import { videoController } from './video/video.controller';
 import fastifyMultipart from 'fastify-multipart';
+import { setupSwagger } from '../docs/setup-swagger';
 
-const server = fastify({ logger: true });
+function ajvPlugin(ajv: any) {
+  ajv.addFormat('binary', { type: 'string', validate: () => true });
```
Comment (JonaszJestem) [inline src/core/index.ts:9]: It does add `binary` format to the input, but it still doesn't render file input in Swagger UI.
That's not the ideal way of doing that, but currently there is no better way as there is still open issue for file uploads:
https://github.com/fastify/fastify-swagger/issues/47

File: src/docs/setup-swagger.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  server.register(fastifySwagger, {
+    routePrefix: '/docs',
+    swagger: {
+      info: {
+        title: 'Image Service',
+        version: require('../../package.json').version,
+      },
+    },
+    exposeRoute: true,
```
Comment (JonaszJestem) [inline src/docs/setup-swagger.ts:14]: Do you want to always expose API docs or expose it for dev environments only?

File: package.json [hunk 3]
```diff
@@ -35,6 +35,7 @@
     "@typescript-eslint/parser": "4.28.3",
     "eslint": "7.31.0",
     "eslint-plugin-import": "2.23.4",
+    "fastify-swagger": "^4.8.4",
```
Comment (KrzysiekWyka) [inline package.json]:
```suggestion
    "fastify-swagger": "4.8.4",
```

File: src/docs/setup-swagger.ts (hunk 2 above)
Comment (KrzysiekWyka) [reply src/docs/setup-swagger.ts:14]: Always

**Review by KrzysiekWyka — CHANGES_REQUESTED** [review]

File: src/core/image/image.controller.ts [hunk 4]
```diff
@@ -5,8 +5,20 @@ import { createCropHandler } from './resize/crop.handler';
 import { createThumbnailHandler } from './resize/thumbnail.handler';
 
 export const imageController = async (fastify: FastifyInstance) => {
-  createMetadataHandler('/metadata', fastify);
-  createThumbnailHandler('/thumbnail', fastify);
-  createCropHandler('/:adapter/crop', fastify);
-  createResizeHandler('/:adapter/resize', fastify);
+  const schema = {
```
Comment (KrzysiekWyka) [inline src/core/image/image.controller.ts]: For me, it is parent schema or base schema. Just schema means nothing.

**Review by KrzysiekWyka — CHANGES_REQUESTED** [review]
Can u rebase it?

File: src/core/image/image.controller.ts (hunk 4 above)
Comment (JonaszJestem) [reply src/core/image/image.controller.ts]: fixed

**Review by KrzysiekWyka — APPROVED** [review]

**Pull request merged**